canard.h

```diff
@@ -30,7 +30,9 @@
 #include <stdint.h>
 #include <stddef.h>
 #include <stdbool.h>
-#include <assert.h>
+
+/// Build configuration header. If necessary, replace it with your own implementation.
+#include "canard_build_config.h"
```

Comment:
Could it be an idea to add a new preprocessor flag which could be provided by the user during compilation in case they want to override the default configs e.g `-D CANARD_ENABLE_CUSTOM_BUILD_CONFIG`? If this flag is provided, then `canard_build_config.h` need to be provided explicitly, otherwise, the default existing configs will be used. 

With the currently proposed solution in this PR, all existing projects using Libcanard will be forced to update their build process, even if they don't need the possibility to override the existing asserts.  By keeping the default behavior as it is today, and instead adding an option to explicit overriding build configs, for projects needing this functionality, this change will remain backward compatible for existing users of Libcanard, meanwhile adding the option of providing a config file for those who need it.

Reply:
> With the currently proposed solution in this PR, all existing projects using Libcanard will be forced to update their build process, even if they don't need the possibility to override the existing asserts.

Is this a significant problem? The change seems trivial -- just adding a single empty header file.

Reply:
I've made the changes as requested. I agree that it is cleaner and less risky to have the user provide the `canard_build_config.h` instead. Regarding using the `CANARD_ENABLE_CUSTOM_BUILD_CONFIG` flag, I like the idea because every user shouldn't  need to provide an empty header file just to accommodate a particular case.